ukis_pysat/raster.py

```diff
@@ -151,14 +151,18 @@ def _pad_to_bbox(self, bbox, mode="constant", constant_values=0):
 
         pad_width = math.ceil(max_diff / self.dataset.transform.to_gdal()[1])  # units / pixel_size
 
+        if pad_width < 0:

```

Comment:
Please check whether we are throwing `ValueError("index can't contain negative values")` of `numpy.pad`, if this is the case it is the correct behaviour in my opinion.

Reply:
As expected, this throws the following error `ValueError: index can't contain negative values`.

Reply:
I agree with you that we should not pad if the pad_width is negative. However, I think we should not throw the error here but simply not pad in such a case. The main aim of the pad option as called by mask (in my opinion) is to ensure that the desired subset is fully covered. If the image is larger than the subset the users get what they want even without padding. if it is smaller we have to pad. Imagine the scenario where one has to align a number of images to the same subset. Some of the images are larger than the subset, others are smaller. I would like the subset method to be smart enough to cover me by recognizing the different cases and acting accordingly to cover the full subset.

Reply:
we could for example do something like this:

```python
        if pad_width > 0:
            # only pad if pad_width > 0
            destination = np.zeros(
                (self.dataset.count, self.__arr.shape[1] + 2 * pad_width, self.__arr.shape[2] + 2 * pad_width,),
                self.__arr.dtype,
            )

            for i in range(0, self.dataset.count):
                destination[i], transform = rasterio.pad(
                    self.__arr[i], self.dataset.transform, pad_width, mode, constant_values=constant_values,
                )

            self.__arr = destination

            self.dataset.close()
            ds = self.__update_dataset(self.dataset.crs, transform, nodata=self.dataset.nodata)

        else:
            ds = self.dataset

        return ds
```

Reply:
With this solution I could live better and I also see the use case, you can just return though. It can be useful for the case you are describing, but this is not very transparent, while an explicit error is.